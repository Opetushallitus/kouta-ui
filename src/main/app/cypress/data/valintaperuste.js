import { merge } from 'lodash';

const withKorkeakouluFields = valintaperuste => merge({}, valintaperuste, {});

export default ({ tyyppi = 'amm' } = {}) => {
  const baseFields = {
    koulutustyyppi: tyyppi,
    id: '649adb37-cd4d-4846-91a9-84b58b90f928',
    tila: 'tallennettu',
    hakutapaKoodiUri: 'hakutapa_0#1',
    kohdejoukkoKoodiUri: 'haunkohdejoukko_0#1',
    nimi: { fi: 'Valintaperusteen nimi' },
    julkinen: false,
    sorakuvausId: '1',
    metadata: {
      koulutustyyppi: tyyppi,
      valintatavat: [
        {
          valintatapaKoodiUri: 'valintatapajono_0#1',
          kuvaus: {},
          nimi: { fi: 'Valintatavan nimi' },
          sisalto: [
            { tyyppi: 'teksti', data: { fi: '<p>Tekstia</p>' } },
            {
              tyyppi: 'taulukko',
              data: {
                nimi: {},
                rows: [
                  {
                    index: 0,
                    isHeader: false,
                    columns: [{ index: 0, text: { fi: 'Solu' } }],
                  },
                ],
              },
            },
          ],
          kaytaMuuntotaulukkoa: false,
          kynnysehto: { fi: 'Kynnysehto' },
          enimmaispisteet: 100.0,
          vahimmaispisteet: 10.0,
        },
      ],
      kielitaitovaatimukset: [],
      kuvaus: { fi: '<p>Loppukuvaus</p>' },
    },
    organisaatioOid: '1.2.246.562.10.594252633210',
    muokkaaja: '1.2.246.562.24.62301161440',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-03T13:56',
    valintakokeidenYleiskuvaus: 'Valintakokeiden kuvaus',
    valintakokeet: [
      {
        nimi: 'Kokeen nimi',
        tietoa: 'Tietoa kokeesta',
        liittyyEnnakkovalmistautumista: true,
        ohjeetEnnakkovalmistautumiseen:
          '<p>Ohjeet ennakkovalmistautumiseen</p>',
        erityisjarjestelytMahdollisia: true,
        ohjeetErityisjarjestelyihin: '<p>Ohjeet erityisj√§rjestelyihin</p>',
        tyyppiKoodiUri: 'tyyppi_1#1',
        tilaisuudet: [
          {
            osoite: {
              osoite: { fi: 'fi osoite', sv: 'sv osoite' },
              postinumeroKoodiUri: 'posti_00350#1',
            },
            aika: {
              alkaa: '2019-04-16T08:44',
              paattyy: '2019-04-18T08:44',
            },
            lisatietoja: {
              fi: 'fi lisatietoja',
              sv: 'sv lisatietoja',
            },
            jarjestamispaikka: 'Auditorio A1',
          },
        ],
      },
    ],
  };

  if (tyyppi === 'amm') {
    return baseFields;
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return withKorkeakouluFields(baseFields);
  }

  return baseFields;
};
