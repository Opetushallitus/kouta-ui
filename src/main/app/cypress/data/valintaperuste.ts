import _ from 'lodash';

const withKorkeakouluFields = valintaperuste =>
  _.merge(
    {
      metadata: {
        valintatavat: [
          {
            valintatapaKoodiUri: 'valintatapajono_tv#1',
            kuvaus: {},
            nimi: { fi: 'Valintatavan nimi' },
            sisalto: [
              { tyyppi: 'teksti', data: { fi: '<p>Tekstia</p>' } },
              {
                tyyppi: 'taulukko',
                data: {
                  nimi: {},
                  rows: [
                    {
                      index: 0,
                      isHeader: false,
                      columns: [{ index: 0, text: { fi: 'Solu' } }],
                    },
                  ],
                },
              },
            ],
            kaytaMuuntotaulukkoa: false,
            kynnysehto: { fi: 'Kynnysehto' },
            enimmaispisteet: 100.02,
            vahimmaispisteet: 10.01,
          },
        ],
      },
    },
    valintaperuste
  );

const valintaperuste = ({ tyyppi = 'amm' } = {}) => {
  const baseFields = {
    koulutustyyppi: tyyppi,
    id: '649adb37-cd4d-4846-91a9-84b58b90f928',
    tila: 'tallennettu',
    hakutapaKoodiUri: 'hakutapa_01#1',
    kohdejoukkoKoodiUri: 'haunkohdejoukko_12#1',
    nimi: { fi: 'Valintaperusteen nimi' },
    julkinen: false,
    sorakuvausId: '1',
    metadata: {
      koulutustyyppi: tyyppi,
      kuvaus: { fi: '<p>Loppukuvaus</p>' },
      sisalto: [
        { tyyppi: 'teksti', data: { fi: '<p>Tekstia</p>' } },
        {
          tyyppi: 'taulukko',
          data: {
            nimi: {},
            rows: [
              {
                index: 0,
                isHeader: false,
                columns: [{ index: 0, text: { fi: 'Solu' } }],
              },
            ],
          },
        },
      ],
      valintakokeidenYleiskuvaus: {
        fi: '<p>Valintakokeiden kuvaus - fi</p>',
      },
    },
    organisaatioOid: '1.2.246.562.10.594252633210',
    muokkaaja: '1.2.246.562.24.62301161440',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-03T13:56',

    valintakokeet: [
      {
        id: 'testi-id-123',
        nimi: {
          fi: 'Kokeen nimi - fi',
        },
        metadata: {
          tietoja: {
            fi: '<p>Tietoa kokeesta - fi</p>',
          },
          liittyyEnnakkovalmistautumista: true,
          ohjeetEnnakkovalmistautumiseen: {
            fi: '<p>Ohjeet ennakkovalmistautumiseen - fi</p>',
          },
          erityisjarjestelytMahdollisia: true,
          ohjeetErityisjarjestelyihin: {
            fi: '<p>Ohjeet erityisj√§rjestelyihin - fi</p>',
          },
          vahimmaispisteet: 10.03,
        },
        tyyppiKoodiUri: 'tyyppi_1#1',
        tilaisuudet: [
          {
            osoite: {
              osoite: { fi: 'fi osoite' },
              postinumeroKoodiUri: 'posti_00350#2',
            },
            aika: {
              alkaa: '2019-04-16T08:44',
              paattyy: '2019-04-18T08:44',
            },
            lisatietoja: {
              fi: 'fi lisatietoja',
            },
            jarjestamispaikka: {
              fi: 'Jarjestamispaikka - fi',
            },
          },
        ],
      },
    ],
  };

  if (tyyppi === 'amm') {
    return baseFields;
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return withKorkeakouluFields(baseFields);
  }

  return baseFields;
};

export default valintaperuste;
