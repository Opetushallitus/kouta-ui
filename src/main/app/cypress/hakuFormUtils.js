import merge from 'lodash/merge';

import organisaatio from './data/organisaatio';
import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
} from './utils';

export const stubHakuFormRoutes = ({ cy, organisaatioOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  stubKoodistoRoute({ koodisto: 'haunkohdejoukko', cy });
  stubKoodistoRoute({ koodisto: 'hakutapa', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'opetuspaikkakk', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });

  cy.route({
    method: 'GET',
    url: '**/haku/list**',
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute({ cy });
  stubOppijanumerorekisteriHenkiloRoute({ cy });
};
