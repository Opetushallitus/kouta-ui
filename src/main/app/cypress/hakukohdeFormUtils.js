import merge from 'lodash/merge';

import { stubKoodistoRoute } from './utils';
import organisaatio from './data/organisaatio';
import haku from './data/haku';

export const stubHakukohdeFormRoutes = ({ cy, organisaatioOid, hakuOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  stubKoodistoRoute({ koodisto: 'pohjakoulutusvaatimustoinenaste', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'valintakokeentyyppi', cy });
  stubKoodistoRoute({ koodisto: 'liitetyypitamm', cy });

  cy.route({
    method: 'GET',
    url: `**/haku/${hakuOid}`,
    response: merge(haku(), {
      oid: hakuOid,
      organisaatioOid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/list**`,
    response: [],
  });
};
