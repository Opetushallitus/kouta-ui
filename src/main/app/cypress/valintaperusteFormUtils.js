import { merge } from 'lodash';

import {
  stubKoodistoRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubKoodiRoute,
  stubCommonRoutes,
} from './utils';

import organisaatio from './data/organisaatio';
import soraKuvaus from './data/soraKuvaus';
import createKoodi from './data/koodi';

export const stubValintaperusteFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url: '**/valintaperuste/list**',
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/sorakuvaus/list**',
    response: [...new Array(10)].map((v, i) =>
      merge(soraKuvaus(), {
        nimi: { fi: `Sora-kuvaus ${i}` },
        id: i.toString(),
        tila: 'julkaistu',
      })
    ),
  });

  stubKoodistoRoute({ koodisto: 'hakutapa' });
  stubKoodistoRoute({ koodisto: 'haunkohdejoukko' });
  stubKoodistoRoute({ koodisto: 'valintatapajono' });
  stubKoodistoRoute({ koodisto: 'kielitaidonosoittaminen' });
  stubKoodistoRoute({ koodisto: 'kielitaitovaatimustyypit' });
  stubKoodistoRoute({ koodisto: 'kielitaitovaatimustyypitkuvaus' });
  stubKoodistoRoute({ koodisto: 'kieli' });
  stubKoodistoRoute({ koodisto: 'osaamistausta' });
  stubKoodistoRoute({ koodisto: 'valintakokeentyyppi' });

  cy.route({
    method: 'GET',
    url: '**/sorakuvaus/1',
    response: merge(soraKuvaus(), {
      nimi: { fi: `Sora-kuvaus 1` },
      id: 1,
      tila: 'julkaistu',
    }),
  });

  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));
  stubKoodiRoute(
    createKoodi({ koodisto: 'posti', koodiArvo: '00350', versio: 1 })
  );

  stubOppijanumerorekisteriHenkiloRoute();
};
