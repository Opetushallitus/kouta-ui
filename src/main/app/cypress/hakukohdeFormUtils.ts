import { merge } from 'lodash';

import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubKoodiRoute,
  stubCommonRoutes,
  jatka,
} from '#/cypress/utils';

import haku from '#/cypress/data/haku';
import hakukohde from '#/cypress/data/hakukohde';
import createKoodi from '#/cypress/data/koodi';
import koulutus from '#/cypress/data/koulutus';
import toteutus from '#/cypress/data/toteutus';
import valintaperuste from '#/cypress/data/valintaperuste';
import { addMockFileRoutes } from './mockUtils';

const toimipisteTarjoajat = [
  '1.2.246.562.10.16538823663',
  '1.2.246.562.10.2013081415065445951365',
  '1.2.246.562.10.20485193278',
  '1.2.246.562.10.20866993056',
  '1.2.246.562.10.23017513880',
  '1.2.246.562.10.45854578546',
  '1.2.246.562.10.55355715099',
  '1.2.246.562.10.55711304158',
  '1.2.246.562.10.56139411567',
  '1.2.246.562.10.60465978314',
  '1.2.246.562.10.656909794910',
  '1.2.246.562.10.78321186062',
  '1.2.246.562.10.879177258610',
  '1.2.246.562.10.93115250668',
];

const selectedToimipisteNimi = /stadin ammatti- ja aikuisopisto, myllypuron toimipaikka/i;

const selectedToimipiste = '1.2.246.562.10.45854578546';

export const fillJarjestyspaikkaSection = ({ jatka: jatkaProp = true }) => {
  cy.findByTestId('jarjestyspaikkaOidSection').within(() => {
    toimipisteTarjoajat.forEach(oid => {
      cy.get(`[value="${oid}"]`).should('exist');
      if (oid === selectedToimipiste) {
        cy.get(`[value="${oid}"]`).should('not.be.disabled');
      } else {
        cy.get(`[value="${oid}"]`).should('be.disabled');
      }
    });

    cy.findByText(selectedToimipisteNimi).click();
    if (jatkaProp) {
      jatka();
    }
  });
};

export const prepareTest = ({
  tyyppi,
  hakuOid,
  hakukohdeOid,
  organisaatioOid,
  edit = false,
  tarjoajat,
}) => {
  const toteutusOid = '2.1.1.1.1.1';
  const koulutusOid = '3.1.1.1.1';
  const valintaperusteId = '649adb37-cd4d-4846-91a9-84b58b90f928';

  const testHakukohdeFields = {
    toteutusOid,
    hakuOid,
    organisaatioOid,
    oid: hakukohdeOid,
    valintaperusteId,
  };

  cy.server();

  addMockFileRoutes('hakukohde.mock.json');
  stubHakukohdeFormRoutes({ organisaatioOid, hakuOid });

  cy.route({
    method: 'GET',
    url: `**/toteutus/${toteutusOid}`,
    response: merge(toteutus({ tyyppi }), {
      oid: toteutusOid,
      organisaatioOid,
      koulutusOid: koulutusOid,
      tila: 'julkaistu',
      tarjoajat,
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/${koulutusOid}`,
    response: merge(koulutus({ tyyppi }), {
      oid: koulutusOid,
      tarjoajat,
      organisaatioOid: organisaatioOid,
      tila: 'julkaistu',
    }),
  });

  cy.route({
    method: 'GET',
    url: '**/valintaperuste/list**',
    response: [
      merge(valintaperuste(), {
        id: valintaperusteId,
        nimi: { fi: 'Valintaperusteen nimi' },
        tila: 'julkaistu',
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url: `**/valintaperuste/${valintaperusteId}`,
    response: merge(valintaperuste(), {
      id: valintaperusteId,
      nimi: { fi: 'Valintaperusteen nimi' },
      tila: 'julkaistu',
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/${hakukohdeOid}`,
    response: merge(hakukohde(), testHakukohdeFields),
  });

  edit
    ? cy
        .route({
          method: 'POST',
          url: '**/hakukohde',
          response: {
            muokattu: false,
          },
        })
        .as('updateHakukohdeRequest')
    : cy
        .route({
          method: 'PUT',
          url: '**/hakukohde',
          response: {
            oid: hakukohdeOid,
          },
        })
        .as('createHakukohdeRequest');

  cy.visit(
    edit
      ? `/organisaatio/${organisaatioOid}/hakukohde/${hakukohdeOid}/muokkaus`
      : `/organisaatio/${organisaatioOid}/toteutus/${toteutusOid}/haku/${hakuOid}/hakukohde`
  );
};

export const stubHakukohdeFormRoutes = ({ organisaatioOid, hakuOid }) => {
  stubCommonRoutes();

  stubKoodistoRoute({ koodisto: 'pohjakoulutusvaatimustoinenaste' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodistoRoute({ koodisto: 'valintakokeentyyppi' });
  stubKoodistoRoute({ koodisto: 'liitetyypitamm' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));
  stubKoodiRoute(
    createKoodi({ koodisto: 'posti', koodiArvo: '00350', versio: 1 })
  );

  cy.route({
    method: 'GET',
    url: `**/haku/${hakuOid}`,
    response: merge(haku(), {
      oid: hakuOid,
      organisaatioOid: organisaatioOid,
      hakulomaketyyppi: 'muu',
      tila: 'julkaistu',
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/list**`,
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute();
  stubOppijanumerorekisteriHenkiloRoute();
};
