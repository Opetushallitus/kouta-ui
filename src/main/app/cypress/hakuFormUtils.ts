import { playMocks } from 'kto-ui-common/cypress/mockUtils';
import { merge } from 'lodash/fp';

import organisaatio from '#/cypress/data/organisaatio';
import organisaatioHierarkia from '#/cypress/data/organisaatioHierarkia';
import hakuMocks from '#/cypress/mocks/haku.mocks.json';

import {
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubCommonRoutes,
} from './utils';

export const stubHakuFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();
  playMocks(hakuMocks);

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**oid=${organisaatioOid}**`,
    },
    { body: organisaatioHierarkia({ rootOid: organisaatioOid }) }
  );

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**`,
    },
    { body: organisaatioHierarkia({ rootOid: organisaatioOid }) }
  );

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    },
    {
      body: merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    }
  );

  cy.intercept(
    {
      method: 'POST',
      url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    },
    {
      body: [
        merge(organisaatio(), {
          oid: organisaatioOid,
        }),
      ],
    }
  );

  cy.intercept({ method: 'GET', url: '**/haku/list**' }, { body: [] });

  cy.intercept({ method: 'GET', url: `**/toteutus/list**` }, { body: [] });

  cy.intercept({ method: 'GET', url: `**/koulutus/list**` }, { body: [] });

  cy.intercept({ method: 'GET', url: `**/search/haku/**` }, { body: [] });

  stubHakemuspalveluLomakkeetRoute();
  stubOppijanumerorekisteriHenkiloRoute();
};
