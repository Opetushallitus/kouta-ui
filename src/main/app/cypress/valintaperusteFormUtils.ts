import { playMocks } from 'kto-ui-common/cypress/mockUtils';
import { merge } from 'lodash/fp';

import valintaperusteMocks from '#/cypress/mocks/valintaperuste.mock.json';

import organisaatio from './data/organisaatio';
import {
  stubOppijanumerorekisteriHenkiloRoute,
  stubCommonRoutes,
} from './utils';

export const stubValintaperusteFormRoutes = ({ organisaatioOid }) => {
  playMocks(valintaperusteMocks);
  stubCommonRoutes();

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    },
    {
      body: merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    }
  );

  cy.intercept(
    {
      method: 'POST',
      url: '**/kouta-backend/organisaatio/organisaatiot',
    },
    {
      body: [
        merge(organisaatio(), {
          oid: organisaatioOid,
        }),
      ],
    }
  );

  cy.intercept(
    { method: 'GET', url: '**/valintaperuste/list**' },
    { body: [] }
  );

  stubOppijanumerorekisteriHenkiloRoute();
};
