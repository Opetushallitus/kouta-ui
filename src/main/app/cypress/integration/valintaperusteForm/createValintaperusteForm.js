import { merge } from 'lodash';
import {
  getRadio,
  selectOption,
  typeToEditor,
  getTableInput,
  getCheckbox,
  chooseKieliversiotLanguages,
  fillKoulutustyyppiSelect,
  fillValintakoeFields,
  jatka,
  getByTestId,
  paste,
} from '#/cypress/utils';
import valintaperuste from '#/cypress/data/valintaperuste';
import { stubValintaperusteFormRoutes } from '#/cypress/valintaperusteFormUtils';

const tallenna = () => {
  getByTestId('tallennaValintaperusteButton').click({
    force: true,
  });
};

const fillKoulutustyyppiSection = path => {
  getByTestId('tyyppiSection').within(() => {
    fillKoulutustyyppiSelect(path);
  });
};

const fillPohjaSection = () => {
  getByTestId('pohjaSection').within(() => {
    jatka();
  });
};

const fillKieliversiotSection = () => {
  getByTestId('kieliversiotSection').within(() => {
    chooseKieliversiotLanguages(['fi']);
  });
};

const fillHakutavanRajausSection = () => {
  getByTestId('hakutapaSection').within(() => {
    getRadio('hakutapa_0#1').click({ force: true });
  });
};

const fillKohdejoukonRajausSection = () => {
  getByTestId('kohdejoukkoSection').within(() => {
    selectOption('haunkohdejoukko_0');
  });
};

const lisaaSisaltoa = tyyppi => {
  getByTestId('sisaltoMenuToggle').click({ force: true });

  getByTestId('sisaltoMenu')
    .first()
    .within(() => {
      if (tyyppi === 'teksti') {
        getByTestId('lisaaTekstia').click({ force: true });
      } else if (tyyppi === 'taulukko') {
        getByTestId('lisaaTaulukko').click({ force: true });
      }
    });
};

const fillValintakoeSection = () => {
  getByTestId('valintakoeSection').within(() => {
    fillValintakoeFields();
    jatka();
  });
};

const fillValintatapaSection = () => {
  getByTestId('valintatavatSection').within(() => {
    getByTestId('valintatapalista').within(() => {
      getByTestId('tapa').within(() => {
        selectOption('valintatapajono_0');
      });

      getByTestId('nimi').find('input').pipe(paste('Valintatavan nimi'));

      getByTestId('sisalto').within(() => {
        lisaaSisaltoa('teksti');

        typeToEditor('Sisältötekstiä');

        lisaaSisaltoa('taulukko');

        getTableInput()
          .find('textarea')
          .invoke('val', '')
          .trigger('paste', {
            clipboardData: {
              getData: () => 'solu1.1\tsolu1.2\rsolu2.1\tsolu2.2',
            },
          });
      });

      getByTestId('kynnysehto').find('textarea').pipe(paste('Kynnysehto'));
      getByTestId('enimmaispistemaara').find('input').pipe(paste('100'));
      getByTestId('vahimmaispistemaara').find('input').pipe(paste('10'));
    });

    jatka();
  });
};

const fillKuvausSection = () => {
  getByTestId('kuvausSection').within(() => {
    getByTestId('nimi').find('input').pipe(paste('Valintaperusteen nimi'));

    getByTestId('kuvaus').within(() => {
      typeToEditor('Kuvaus');
    });

    jatka();
  });
};

const fillSoraKuvausSection = () => {
  getByTestId('soraKuvausSection').within(() => {
    selectOption('Sora-kuvaus 1');

    jatka();
  });
};

const fillJulkisuusSection = () => {
  getByTestId('julkinenSection').within(() => {
    getCheckbox(null).check({ force: true });
    jatka();
  });
};

const fillPerustiedotSection = () => {
  getByTestId('perustiedotSection').within(() => {
    fillKoulutustyyppiSection(['korkeakoulutus']);
    fillKieliversiotSection();
    fillHakutavanRajausSection();
    fillKohdejoukonRajausSection();

    jatka();
  });
};

const fillTilaSection = (tila = 'julkaistu') => {
  getByTestId('tilaSection').within(() => {
    getRadio(tila).check({ force: true });
  });
};

describe('createValintaperusteForm', () => {
  const organisaatioOid = '1.1.1.1.1.1';
  const createdValintaperusteId = '1.2.3.4.5.6';

  beforeEach(() => {
    cy.server();
    stubValintaperusteFormRoutes({ organisaatioOid });

    cy.visit(`/organisaatio/${organisaatioOid}/valintaperusteet/kielivalinnat`);
  });

  it('should be able to create valintaperuste', () => {
    cy.route({
      method: 'GET',
      url: `**/valintaperuste/${createdValintaperusteId}`,
      response: [
        merge(valintaperuste(), {
          oid: createdValintaperusteId,
        }),
      ],
    });

    cy.route({
      method: 'PUT',
      url: '**/valintaperuste',
      response: {
        id: createdValintaperusteId,
      },
    }).as('createValintaperusteRequest');

    fillPerustiedotSection();
    fillPohjaSection();
    fillKuvausSection();
    fillValintatapaSection();
    fillValintakoeSection();
    fillSoraKuvausSection();
    fillJulkisuusSection();
    fillTilaSection();

    tallenna();

    cy.wait('@createValintaperusteRequest').then(({ request }) => {
      cy.wrap(request.body).snapshot();
    });

    cy.location('pathname').should(
      'eq',
      `/kouta/organisaatio/${organisaatioOid}/valintaperusteet/${createdValintaperusteId}/muokkaus`
    );
  });
});
