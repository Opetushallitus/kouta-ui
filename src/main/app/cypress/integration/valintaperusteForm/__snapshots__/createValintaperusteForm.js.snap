exports[`createValintaperusteForm > should be able to create valintaperuste #0`] =
{
  "hakutapaKoodiUri": "hakutapa_0#1",
  "kielivalinta": [
    "fi"
  ],
  "kohdejoukkoKoodiUri": "haunkohdejoukko_0#1",
  "koulutustyyppi": "yo",
  "metadata": {
    "kielitaitovaatimukset": [],
    "kuvaus": {
      "fi": "<p>Kuvaus</p>"
    },
    "osaamistaustaKoodiUrit": [],
    "tyyppi": "yo",
    "valintatavat": [
      {
        "enimmaispisteet": 100,
        "kaytaMuuntotaulukkoa": false,
        "kuvaus": {},
        "kynnysehto": {
          "fi": "Kynnysehto"
        },
        "nimi": {
          "fi": "Valintatavan nimi"
        },
        "sisalto": [
          {
            "data": {
              "fi": "<p>Sisältötekstiä</p>"
            },
            "tyyppi": "teksti"
          },
          {
            "data": {
              "rows": [
                {
                  "columns": [
                    {
                      "index": 0,
                      "text": {
                        "fi": "solu1.1"
                      }
                    },
                    {
                      "index": 1,
                      "text": {
                        "fi": "solu1.2"
                      }
                    }
                  ],
                  "index": 0
                },
                {
                  "columns": [
                    {
                      "index": 0,
                      "text": {
                        "fi": "solu2.1"
                      }
                    },
                    {
                      "index": 1,
                      "text": {
                        "fi": "solu2.2"
                      }
                    }
                  ],
                  "index": 1
                }
              ]
            },
            "tyyppi": "taulukko"
          }
        ],
        "vahimmaispisteet": 10,
        "valintatapaKoodiUri": "valintatapajono_0#1"
      }
    ]
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Valintaperusteen nimi"
  },
  "onkoJulkinen": true,
  "organisaatioOid": "1.1.1.1.1.1",
  "sorakuvausId": "1",
  "tila": "julkaistu",
  "valintakokeet": [
    {
      "erityisjarjestelytMahdollisia": true,
      "liittyyEnnakkovalmistautumista": true,
      "nimi": {
        "fi": "nimi"
      },
      "ohjeetEnnakkovalmistautumiseen": {
        "fi": "<p>ohjeet ennakkovalmistautumiseen</p>"
      },
      "ohjeetErityisjarjestelyihin": {
        "fi": "<p>ohjeet erityisjärjestelyihin</p>"
      },
      "tietoja": {
        "fi": "<p>Tietoa hakijalle</p>"
      },
      "tilaisuudet": [
        {
          "aika": {
            "alkaa": "2019-04-02T10:45",
            "paattyy": "2019-04-02T19:00"
          },
          "jarjestamispaikka": {
            "fi": "paikka"
          },
          "lisatietoja": {
            "fi": "lisatietoja"
          },
          "osoite": {
            "osoite": {
              "fi": "osoite"
            },
            "postinumeroKoodiUri": "posti_0#2"
          }
        }
      ],
      "tyyppiKoodiUri": "valintakokeentyyppi_1#1"
    }
  ],
  "valintakokeidenYleiskuvaus": {
    "fi": "<p>Valintakokeiden kuvaus</p>"
  }
};
